components/shared/collection/PriceChart.vue

```diff
@@ -0,0 +1,199 @@
+  protected options = {}
+
+  protected resetZoom() {
+    this.$emit('resetZoom')
+  }
+
+  protected async mounted() {
+    this.priceChart()
+  }
+
+  protected priceChart() {
+    if (this.priceData[0]?.length || this.priceData[1]?.length) {
+      const median = getCollectionMedian(this.priceData[1])
+      const labelsListing = this.priceData[0].map(getLabel)
+      const labelsBuy = this.priceData[1].map(getLabel)
+
+      this.labels = [...new Set([...labelsListing, ...labelsBuy])]
+      this.datasets = [
+        {
+          label: 'Floor Price',
+          data: getCollectionChartData(this.priceData[0]) as any,
+          borderColor: '#d32e79',
+          ...baseLineOptions,
+        },
+        {
+          label: 'Sold NFT Price',
+          data: getCollectionChartData(this.priceData[1]),
+          borderColor: '#00BB7F',
+          ...baseLineOptions,
+        },
+      ]
+
+      if (this.priceData[1][0]?.average) {
+        this.datasets.push({
+          label: 'Trailing Average',
+          data: getMovingAverage(
+            getCollectionChartData(this.priceData[1], mapToAverage)
+          ),
+          borderColor: 'yellow',
+          ...baseLineOptions,
+        })
+      }
+
+      this.options = {
+        maintainAspectRatio: false,
+        plugins: {
+          tooltip: {
+            callbacks: {
+              afterLabel: ({ dataIndex, dataset }) => {
+                return `Count: ${dataset.data[dataIndex]?.count || 0}`
+              },
+            },
+          },
+          annotation: {
+            annotations: {
+              median: {
+                type: 'line',
+                yMin: median,
+                yMax: median,
+                borderColor: '#00BB7F',
+                borderWidth: 2,
+                borderDash: [10, 5],
+              },
+            },
+          },
+          zoom: {
+            pan: {
+              enabled: false,
+            },
+            zoom: {
+              drag: {
+                enabled: true,
+                backgroundColor: '',
+              },
+              pinch: {
+                enabled: true,
+              },
+              mode: 'xy',
+              onZoomComplete({ chart }) {
+                chart.update('none')
+              },
+            },
+          },
+        },
+        scales: {
+          x: {
+            type: 'time',
+            time: {
+              unit: 'day',
+            },
+            ticks: {
+              maxRotation: 0,
+              minRotation: 0,
+              color: '#fff',
+            },
+          },
+          y: {
+            ticks: {
+              callback: (value) => {
+                return `${Number(value).toFixed(2)} ${this.unit}`
+              },
+              maxTicksLimit: 7,
+              color: '#fff',
+            },
+            grid: {
+              color: '#3a3a3a',
+            },
+          },
+        },
+      }
+    }
+  }
+
+  @Watch('priceData')
+  async watchData() {
```

Comment:
async but no await on call? is `priceChart()` even a asynchronous function?

Reply:
oh you're right, updated 👍